flocker/acceptance/test_restarts.py

```diff
@@ -0,0 +1,109 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Tests for restarting and reboots and their interactions.
+"""
+
+import os
+
+from subprocess import call
+from unittest import SkipTest
+
+from twisted.python.filepath import FilePath
+from twisted.trial.unittest import TestCase
+
+from treq import get, content
+
+from ..testtools import loop_until, random_name
+from .testtools import create_dataset, require_cluster
+
+
+REBOOT_SERVER = FilePath(__file__).sibling(b"reboot_httpserver.py")
+
+
+class RestartTests(TestCase):
+    """
+    Tests for restart policies.
+    """
+    @require_cluster(1)
+    def test_restart_always_reboot_with_dataset(self, cluster):
+        """
+        If a container has a restart policy of ``always`` and a volume mapped
+        to a dataset, on reboots it will only be restarted after the
+        volume becomes available.
+        """
+        if not os.environ.get("RUN_REBOOT_TESTS"):
+            raise SkipTest(
+                    "Don't want to run this on buildbot, for now at least.")
+        node = cluster.nodes[0]
+        # Implicitly uses first node:
+        creating_dataset = create_dataset(self, cluster)
+
+        def query_server():
+            req = get(
+                "http://{host}:12345".format(host=node.public_address),
+                persistent=False
+            ).addCallbacks(content)
+            return req
+
+        def created_dataset(dataset):
+            print "DATASET", dataset
+            http_server = {
+                u"name": random_name(self),
+                u"node_uuid": str(cluster.nodes[0].uuid),
+                u"image": u"python:2.7-slim",
+                u"ports": [{u"internal": 12345, u"external": 12345}],
+                # We expect restarts to occur after reboot with this policy:
+                u'restart_policy': {u'name': u'always'},
```

Comment:
:question: The restart policy will be discarded by the container API, right? It confused me for a moment. Let's remove it.